include/aspect/utilities.h

```diff
@@ -813,6 +813,42 @@ namespace aspect
          */
         std_cxx11::unique_ptr<aspect::Utilities::AsciiDataLookup<1> > lookup;
     };
+
+    /**
+     * A weighted p norm average caluclator:
+     * $\text{weighed\_p\_norm\_average} =
+     * \frac{1}{k}(\sum_{i=1}^k \big(\text{weights}_i \text{values}_i^p)\big)^{\frac{1}{p}}$.
+     *
+     * There have been several special cases implemented, mostly for performance reasons, and
```

Comment:
Write this as "This function implements several special cases..."

I don't understand what the "performance reasons" are -- are you suggesting that it would have been inefficient to implement them outside the function?

In any case, you only document two of the special cases, but how is one to learn about the others?